.github/workflows/library-artifacts.yaml

```diff
@@ -0,0 +1,105 @@
+name: artifacts
+
+on:
+  pull_request:
+    paths-ignore:
+      - "apis/**"
+      - ".pre-commit-config.yaml"
+  push:
+    branches: [main, master]
+  release:
+
+env:
+  TILEDBSOMA_VER: "1.2.4"
```

Comment:
@eddelbuettel -- an idea from @Shelnutt2 (conversationally) -- we can get these from github variables
https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
`GITHUB_REF` and `GITHUB_SHA`

Reply:
Oh yes sure -- he and I chatted today too. But this all can rest til 1.0R is shipped and then we clean
- where/how to get release and sha from
- how tickle (releases only ?)
- properly sequence (as the library needs to be a pre-requirement for the R package)

I think it can rest til then.

(And note that when I built / tested this I had no release tags to work off so it's all "in flux" in that way too.)

Reply:
So "your requesting change" is not all that meaningful.  Or feel free to self-assign.  I did what I could here.

The remainder has external dependencies exogeneous to this PR.

Reply:
That has been done ... but note that the Action now _depends_ on two things

- triggered by a release (i.e. when a release is made and tagged, the action runs and provides the artifacts)
- this _requires_ a PAT for access, we may have to add one / align the name we refer to

Reply:
@eddelbuettel re the first: I think we can add this

```
on:
  #pull_request:
  #  paths-ignore:
  #    - "apis/**"
  #    - ".pre-commit-config.yaml"
  #push:
  #  branches: [main, master]
  release:
    types:
      - created
  workflow_dispatch: # add this
```

Then it will be manually launchable in the GitHub web UI

Reply:
and re the PAT -- IIUC we can re-use the same one from other actions ...

Reply:
Good one re 'workflow_dispatch' -- I had not used that, but it makes sense.  The main one here is to turn _off_ all the usual action on pushes, merges, ... 

Also re the PAT I had not even check but I wired one up in my test repo so I would only need to carry its identifier over into this.

Reply:
@johnkerl Done for the workflow_dispatch.  I see no existing PAT so we'd have to get one in.